### PR TITLE
including jwt token in browsers local storage

### PR DESCRIPTION
**Pull request by harshaellanki**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [ ] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/n/projects/2640063/stories/186571692

# A brief description of the changes

Current behavior:

New behavior:

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable that is used to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

### CODE REVIEW

File: app/javascript/legacy/external_applications.js [hunk 1]
```diff
@@ -1,17 +1,20 @@
 function setupExternalRedirectedListener() {
   console.log('setupExternalRedirectedListener');
   var changeElement = document.getElementById("external_app_redirection_data");
-  console.log(changeElement)
+  console.log(changeElement);
```
Comment (polographer) [inline app/javascript/legacy/external_applications.js:4]: can you remove this console.log

File: app/javascript/legacy/external_applications.js [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
+
   if (changeElement != null) {
     var redirUrl = changeElement.getAttribute("data-redirect-url");
     var jwtVal = changeElement.getAttribute("data-redirect-jwt");
+
     if (redirUrl != null) {
       window.localStorage.setItem("jwt", jwtVal);
-      window.location.assign(redirUrl + "?auth_token=" + jwtVal);
+      window.localStorage.setItem("token", jwtVal);
```
Comment (polographer) [inline app/javascript/legacy/external_applications.js:12]: for some reason this line is flagged by codesweep but not the previous one, can we reuse the previous one ?

**Review by polographer — CHANGES_REQUESTED** [review]

**Review by polographer — DISMISSED** [review]

File: app/views/ui-components/v1/navs/_primary_nav.html.slim [hunk 3]
```diff
@@ -92,3 +93,17 @@
               li.pl-1 role="presentation" class="#{pundit_class HbxProfile, :view_admin_tabs?}"
                 = link_to(notifier.notice_kinds_path, "aria-expanded" => "true", "aria-controls" => "notices", "role" => "tab" ) do
                   | Notices
+
+
+javascript:
+  document.addEventListener('DOMContentLoaded', function() {
+    window.localStorage.removeItem("jwt");
+    var agencyStaffLink = document.getElementById('agencyStaffLinkToExternal');
```
Comment (bbodine1) [inline app/views/ui-components/v1/navs/_primary_nav.html.slim:101]: Use `const` over `var`
- explanation: `var` will check outside of the scope of the current function for set values. `const` will scope the variable to the current function. 99% of the time you would never have an issue with a var scope leaking but with `const` you never will.

File: app/views/ui-components/v1/navs/_primary_nav.html.slim [hunk 4]
```diff
@@ -92,3 +93,17 @@
...
+    agencyStaffLink.addEventListener('click', function(event) {
+      event.preventDefault();
+      var jwtToken = "<%= jwt_for_external_application %>";
```
Comment (bbodine1) [inline app/views/ui-components/v1/navs/_primary_nav.html.slim:104]: Use `const` over `var`

File: app/views/ui-components/v1/navs/_primary_nav.html.slim [hunk 5]
```diff
@@ -92,3 +93,17 @@
               li.pl-1 role="presentation" class="#{pundit_class HbxProfile, :view_admin_tabs?}"
                 = link_to(notifier.notice_kinds_path, "aria-expanded" => "true", "aria-controls" => "notices", "role" => "tab" ) do
                   | Notices
+
+
+javascript:
+  document.addEventListener('DOMContentLoaded', function() {
+    window.localStorage.removeItem("jwt");
+    var agencyStaffLink = document.getElementById('agencyStaffLinkToExternal');
+    agencyStaffLink.addEventListener('click', function(event) {
```
Comment (bbodine1) [inline app/views/ui-components/v1/navs/_primary_nav.html.slim:102]: Do we have to wait for a click to set the localstorage jwt? Could we set it as soon as the DOM ContentLoaded is fired?

**Review by nisanthyaganti9 — DISMISSED** [review]

**Review by bbodine1 — COMMENTED** [review]
Comments on file below

**Review by bbodine1 — APPROVED** [review]

**Review by polographer — APPROVED** [review]

**Pull request merged**